server/usermanager.ts

```diff
@@ -744,28 +750,20 @@ const accountRecoveryStart: RequestHandler<
 	OttResponseBody<{}>,
 	OttApiRequestAccountRecoveryStart
 > = async (req, res) => {
+	const body = OttApiRequestAccountRecoveryStartSchema.parse(req.body);
 	if (conf.get("rate_limit.enabled")) {
 		await consumeRateLimitPoints(res, req.ip, 100);
 	}
 
-	if (!("email" in req.body) && !("username" in req.body)) {
-		throw new BadApiArgumentException(
-			"email,username",
-			"Either email or username is required."
-		);
-	}
-
 	const query = {
-		user: req.body.email ?? req.body.username,
+		user: (body as { email?: string }).email ?? (body as { username?: string })?.username,
 	};
 	const user = await getUser(query);
 
-	if (!user.email) {
-		throw new NoEmail();
+	if (user.email) {
+		(await sendPasswordResetEmail(user.email)).unwrap();
 	}
 
-	(await sendPasswordResetEmail(user.email)).unwrap();
-
```

Comment:
Try not to change the functionality of these endpoints. This makes the endpoint fake succeed when there is no email on the account.

This logic needs to remain as is because its possible for an account to have no email associated with it (like if they just use a social login).